### PR TITLE
feat(containerization): enhance docker compose config

### PR DESCRIPTION
**Pull request by KadlagAkash**
### Description
This PR addresses issue #153, aiming to enhance developer experience with Docker Compose improvements.

### Changes Made:

- Implement Docker Compose watch feature for automatic sync and rebuild of backend service.
- Optimize MongoDB service to reduce log clutter.

### Testing:

- Manually tested synching and rebuilding actions.
- Checked for any unintended side effects or regressions.

### Additional Notes:
- The code has been reviewed and follows the coding standards and conventions of the project.
- [Docker Compose Watch](https://docs.docker.com/compose/file-watch/)
- [Hands on](https://www.youtube.com/live/I-htDVxmFGM?feature=shared)

### Related Issue:

- Issue #153 

### Screenshots:

After running `docker compose up --build`:

- ### Press `w` to Enable Watch 
   Screenshot 2024-05-02 at 11 27 50

- ### Syncing `backend` service after changes in code
  Screenshot 2024-05-02 at 12 28 47
  
- ### Rebuilding `backend` service after changes in dependencies
   Screenshot 2024-05-02 at 11 31 16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,10 +90,10 @@ To run the FreeAPI project, follow these steps:
 5. Run the Docker Compose command:
 
 ```bash
-docker-compose up --build --attach backend
+docker-compose up --build --watch
```
Comment (wajeshubham) [inline README.md:93]: `unknown flag: --watch` getting this error

**Review by wajeshubham — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (KadlagAkash) [reply README.md:93]: re-iterating on it

**Comment by KadlagAkash** [comment]
@wajeshubham --watch is available after Docker compose 2.25.0 release

Ref: https://docs.docker.com/compose/release-notes/#2250

**Comment by wajeshubham** [comment]
We want this project to be backward compatible and must work on majority of versions of the tools that we are using. I guess --watch flag is not needed and we can just use --attach to ignore db logs. For now, I'm closing this PR but we will consider this PR in future. Thank you

**Comment by KadlagAkash** [comment]
@wajeshubham  

I appreciate your consideration of my PR, and I’m happy to revisit these changes in the future when it aligns better with our project goals.

Thank you for reviewing and providing your insights.

**Pull request closed without merging**